src/main/java/org/folio/circulation/domain/anonymization/DefaultLoanAnonymizationService.java

```diff
@@ -1,59 +1,46 @@
+import org.folio.circulation.domain.anonymization.service.LoanAnonymizationFinderService;
+import org.folio.circulation.support.Clients;
 import org.folio.circulation.support.Result;
 
-/**
- * Checks loan eligibility for anonymization.
- * By default a loan can only be anonymized if it's closed and there are no fees
- * and fines associated with it.
- *
- */
 public class DefaultLoanAnonymizationService implements LoanAnonymizationService {
 
-  private static final String CAN_BE_ANONYMIZED_KEY = "_";
-
-  private final LoanAnonymizationHelper anonymization;
   private final AnonymizeStorageLoansRepository anonymizeStorageLoansRepository;
-
-  DefaultLoanAnonymizationService(LoanAnonymizationHelper anonymization) {
-    this.anonymization = anonymization;
-    anonymizeStorageLoansRepository = new AnonymizeStorageLoansRepository(anonymization.clients());
+  private final AnonymizationCheckersFacade anonymizationCheckersFacade;
+  private final LoanAnonymizationFinderService loansFinder;
+
+  DefaultLoanAnonymizationService(Clients clients, AnonymizationCheckersFacade anonymizationCheckersFacade,
+      LoanAnonymizationFinderService loansFinderService) {
+    this.anonymizationCheckersFacade = anonymizationCheckersFacade;
+    this.loansFinder = loansFinderService;
+    anonymizeStorageLoansRepository = new AnonymizeStorageLoansRepository(clients);
   }
 
   @Override
   public CompletableFuture<Result<LoanAnonymizationRecords>> anonymizeLoans() {
 
-    return anonymization.loansFinder().findLoansToAnonymize()
+    return loansFinder.findLoansToAnonymize()
       .thenApply(r -> r.map(new LoanAnonymizationRecords()::withLoansFound))
-      .thenCompose(this::segregateLoans)
+      .thenCompose(this::segregateLoanRecords)
       .thenCompose(r -> r.after(anonymizeStorageLoansRepository::postAnonymizeStorageLoans));
   }
 
-  private CompletableFuture<Result<LoanAnonymizationRecords>> segregateLoans(
+  private CompletableFuture<Result<LoanAnonymizationRecords>> segregateLoanRecords(
       Result<LoanAnonymizationRecords> anonymizationRecords) {
 
     return completedFuture(anonymizationRecords.map(records -> {
-      HashSetValuedHashMap<String, String> multiMap = new HashSetValuedHashMap<>();
-      for (Loan loan : records.getLoansFound()) {
-        for (AnonymizationChecker checker : anonymization.anonymizationCheckers()) {
-          if (!checker.canBeAnonymized(loan)) {
-            multiMap.put(checker.getReason(), loan.getId());
-          } else {
-            multiMap.put(CAN_BE_ANONYMIZED_KEY, loan.getId());
-          }
-        }
-      }
-
-      return records.withAnonymizedLoans(multiMap.remove(CAN_BE_ANONYMIZED_KEY))
-        .withNotAnonymizedLoans(multiMap);
+      HashSetValuedHashMap<String, String> segregatedLoans = anonymizationCheckersFacade.segregateLoans(records.getLoansFound());
+      return records.withAnonymizedLoans(segregatedLoans.remove(CAN_BE_ANONYMIZED_KEY))
+        .withNotAnonymizedLoans(segregatedLoans);
```

Comment:
It is to gather all loans that pass all the checkers in a single list - that list is used later on in a request body for a request to anonymize storage loans.

Reply:
Why is a magic value needed rather than accumulating the relevant loans into two new lists?

Reply:
This is to move around loans that are eligible and not eligible for anonymization in a single structure. Later on, its used in the loan anonymization records.

Reply:
Could that structure have two lists instead, rather than a single list with a magic discriminator?

I don't understand why a single list with a magic discriminator describes the intent of the segregation better.